qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -62,6 +62,28 @@ def init():
     if webenginesettings.private_profile:
         _qute_scheme_handler.install(webenginesettings.private_profile)
 
+    should_use_dbus = (
+        qtutils.version_check("5.13") and
+        config.val.content.notification_presenter == "libnotify" and
+        # Don't even try to use DBus notifications on platforms that won't have
+        # it supported.
+        not utils.is_windows and
+        not utils.is_mac
+    )
+    if should_use_dbus:
+        # Import here because some of the classes it uses don't even exist
+        # prior to 5.13.
+        from qutebrowser.browser.webengine import notification
+        log.init.debug("Setting up DBus notification presenter...")
+        try:
+            presenter = notification.DBusNotificationPresenter()
+            for p in [webenginesettings.default_profile, webenginesettings.private_profile]:
+                if not p:
+                    continue
+                presenter.install(p)
+        except notification.DBusException as e:
+            log.init.error("Failed to initialize DBus notification presenter: {}".format(e))
+
```

Comment:
Hmm, I think it would be good to move this code into a `notification.init()` function instead, which could then be called from here (as there's a bit more logic here which would fit better inside `notification.py`). I think if you move the `QWebEngineNotification` import under a `if typing.TYPE_CHECKING:` block and use `'QWebEngineNotification'` as a string in the type annotations, it should be fine to import the file here anyways, right?

Reply:
Good idea. I moved this block into `notification.init` and fixed everything so the module should be safe to import.